File: bugbank-selenium/src/test/java/bugbank/pages/RegisterPage.java
package bugbank.pages;

import org.openqa.selenium.*;
import org.openqa.selenium.support.ui.ExpectedConditions;

import java.util.List;
import java.util.Objects;

public class RegisterPage extends BasePage {

    private static final By campoEmail = By.cssSelector("div.card__register input[type='email']");
    private static final By emailMenssagemErro = By.cssSelector("div.card__register input[type='email'] + p");
    private static final By campoName = By.cssSelector("input[type='name']");
    private static final By nameMenssagemErro = By.cssSelector("input[type='name'] + p");
    private static final By campoPassword = By.cssSelector("div.card__register input[type='password']");
    private static final By passwordMenssagemErro = By.cssSelector("div.card__register input[type='password'] + p");
    private static final By campoPasswordConfirm = By.cssSelector("div.card__register input[name='passwordConfirmation']");
    private static final By passwordConfirmMenssagemErro = By.cssSelector("div.card__register input[name='passwordConfirmation'] + p");
    private static final By checkContaComSaldo = By.id("toggleAddBalance");
    private static final By botaoCadastrar = By.cssSelector("div.card__register > form > button[type=submit]");
    private static final By textoModal = By.id("modalText");
    private static final By botaoFecharModal = By.id("btnCloseModal");
    private static final By botaoRegistrar = By.cssSelector("div.login__buttons > button[type=button]");
    private static final String URL = "https://bugbank.netlify.app/";


    public static void acessarPagina() {
        driver.get(URL);
    }
    public static void registrarUsuarioComSaldo(String email, String name, String password, String passwordConfirm) {
        abrirFormularioDeRegistro();
        preencherCampoEmail(email);
        preencherCampoName(name);
        preencherCampoPassword(password);
        preencherCampoPasswordConfirm(passwordConfirm);
        clicarCheckContaComSaldo();
        clicarBotaoCadastrar();
    }

    public static void registrarUsuarioSemSaldo(String email, String name, String password, String passwordConfirm) {
        abrirFormularioDeRegistro();
        preencherCampoEmail(email);
        preencherCampoName(name);
        preencherCampoPassword(password);
        preencherCampoPasswordConfirm(passwordConfirm);
        clicarBotaoCadastrar();
    }


    public static void abrirFormularioDeRegistro() {
        acessarPagina();
        clickarBotaoDeRegistro();
    }

    public static void clickarBotaoDeRegistro() {
        click(botaoRegistrar);
    }
    public static void preencherCampoEmail(String email) {
        sendKeys(campoEmail, email);
    }

    public static void preencherCampoName(String name) {
        sendKeys(campoName, name);
    }

    public static void preencherCampoPassword(String password) {
        sendKeys(campoPassword, password);
    }

    public static void preencherCampoPasswordConfirm(String passwordConfirm) {
        sendKeys(campoPasswordConfirm, passwordConfirm);
        sendKeys(campoPasswordConfirm, Keys.TAB);
    }

    public static void clicarCheckContaComSaldo() {

        JavascriptExecutor js = (JavascriptExecutor) driver;
        js.executeScript("arguments[0].click();", driver.findElement(checkContaComSaldo));

    }

    public static void clicarBotaoCadastrar() {
        click(botaoCadastrar);
    }

    public static String getTextoModal() {
        wait.until(ExpectedConditions.visibilityOfElementLocated(textoModal));
        return getText(textoModal);
    }

    public static void fecharModal() {
        click(botaoFecharModal);
    }

    /*public static List<WebElement> getMensagensDeErro() {
        return driver.findElement(cardRegister).findElements(mensagensErro);
    }*/

    public static String getMensagemDeEmail() {
        return getWithoutWaitText(emailMenssagemErro);
    }

    public static String getMensagemDeName() {
        return getWithoutWaitText(nameMenssagemErro);
    }

    public static String getMensagemDePassword() {
        return getWithoutWaitText(passwordMenssagemErro);
    }

    public static String getMensagemDePasswordConfirm() {
        return getWithoutWaitText(passwordConfirmMenssagemErro);
    }



}

File: bugbank-cypress/cypress/e2e/pages/register-page.js
 

export class RegisterPage {

    static url = 'https://bugbank.netlify.app/';
    static registerButton = "div.login__buttons > button[type=button]";
    static emailInput = "div.card__register input[type='email']";
    static nameInput = "input[type='name']";
    static passwordInput = "div.card__register input[name='password']";
    static passwordConfirmInput = "div.card__register input[name='passwordConfirmation']";
    static checkContaComSaldo = "#toggleAddBalance";
    static cadastroButton = "div.card__register > form > button[type=submit]";
    static modalText = "#modalText";
    static btnCloseModal = "#btnCloseModal";
    static mensagemDeEmail = "div.card__register input[type='email'] + p";
    static mensagemDeName = "input[type='name'] + p";
    static mensagemDePassword = "div.card__register input[name='password'] + p";
    static mensagemDePasswordConfirm = "div.card__register input[name='passwordConfirmation'] + p";

    

    static registrarUsuarioComSaldo(email, name, password, passwordConfirm) {
        this.abrirFormularioDeRegistro();
        this.preencherCampoEmail(email);
        this.preencherCampoName(name);
        this.preencherCampoPassword(password);
        this.preencherCampoPasswordConfirm(passwordConfirm);
        this.clicarCheckContaComSaldo();
        this.clicarBotaoCadastrar();
    }

    static registrarUsuarioSemSaldo(email, name, password, passwordConfirm) {
        this.abrirFormularioDeRegistro();
        this.preencherCampoEmail(email);
        this.preencherCampoName(name);
        this.preencherCampoPassword(password);
        this.preencherCampoPasswordConfirm(passwordConfirm);
        this.clicarBotaoCadastrar();
    }

    static acessarPagina() {
        cy.visit(this.url);
    }

    static abrirFormularioDeRegistro() {
        cy.get(this.registerButton).click();
    }

    static preencherCampoEmail(email) {
        cy.get(this.emailInput).type(email, {
            force: true,            
        });
    }

    static preencherCampoName(name) {
        cy.get(this.nameInput).type(name, {
            force: true,
        });
    }

    static preencherCampoPassword(password) {
        cy.get(this.passwordInput).type(password, {
            force: true,
        });
    }

    static preencherCampoPasswordConfirm(passwordConfirm) {
        cy.get(this.passwordConfirmInput).type(passwordConfirm, {
            force: true,
        }).tab();
    }

    static clicarCheckContaComSaldo() {
        cy.get(this.checkContaComSaldo).click({ force: true });
    }

    static clicarBotaoCadastrar() {
        cy.get(this.cadastroButton).click({ force: true });
    }

    static getTextoModal() {
        cy.get(this.modalText).should('be.visible');
        return cy.get(this.modalText).invoke('text');
    }

    static fecharModal() {
        cy.get(this.btnCloseModal).click();
    }

    static getMensagemDeEmail = () => {
        return cy.get(this.mensagemDeEmail).invoke('text');
    }

    static getMensagemDeName = () => {
        return cy.get(this.mensagemDeName).invoke('text');
    }

    static getMensagemDePassword = () => {
        return cy.get(this.mensagemDePassword).invoke('text');
    }

    static getMensagemDePasswordConfirm = () => {
        return cy.get(this.mensagemDePasswordConfirm).invoke('text');
    }
}


